use-cases.md

```diff
@@ -0,0 +1,184 @@
+
+```python
+    def traced_request(request, operation, http_client):
+        # retrieve current span from propagated request context
+        parent_span = get_current_span()
+
+        # start a new child span or a brand new trace if no parent
+        if parent_span is None:
+            span = tracer.start_trace(operation_name=operation)
+        else:
+            span = parent_span.start_child(operation_name=operation)
+        span.set_tag('server.http.url', request.full_url)
+
+        # encode Trace Context into HTTP request headers
+        h_ctx, h_attr = tracer.trace_context_to_text(
+            trace_context=span.trace_context)
+
+        for key, value in h_ctx.iteritems():
+            request.add_header(key, value)
+        if h_attr:
+            for key, value in h_attr.iteritems():
+                request.add_header(key, value)
+
+        # define a callback where we can finish the span 
+        def on_done(future):
+            if future.exception():
+                span.log_event_with_payload('exception', exception)
+            else:
+                span.set_tag('http.status_code', future.result().status_code)
+            span.finish()
+
+        try:
+            future = http_client.execute(request)
+            future.add_done_callback(on_done)
+            return future
+        except Exception e:
+            span.log_event_with_payload('exception', e)
+            span.finish()
+            raise
+```
+
+  * The `get_current_span()` function is not a part of the OpenTracing API. It is meant to represent some util method of retrieving the current span from the current request context propagated implicitly (as is often the case in Python).
+  * The encoding function `trace_context_to_text` returns two maps, one representing the encoding of the trace context identity, the other the trace context attributes. We copy both maps into the HTTP request headers.
+  * We assume the HTTP client is asynchronous, so it returns a Future, and we need to add an on-completion callback to be able to finish the current child span.
+  * If HTTP client returns a future with exception, we log the exception to the span with `log_event_with_payload` method.
+  * Because the HTTP client may throw an exception even before returning a Future, we use a try/catch block to ensure that we finish the span in all circumstances.
+
+
+### Using Distributed Context Propagation
+
+The client and server examples above took care of propagating Trace Context on the wire, which includes any Trace Attributes.  The client may use the Trace Attributes to pass additional data to the server and any other downstream server it might call.
+
+```python
+
+    # client side
+    span.trace_context.set_attribute('auth-token', '.....')
+
+    # server side (one or more levels down from the client)
+    token = span.trace_context.get_attribute('auth-token')
+```
```

Comment:
I think the example of auth-token is an unnecessarily scary one. Also, this assumes the primary context propagation api is OpenTracing, when in reality many frameworks have one to do things like this directly. I agree we should use an example more likely for a user to call, or at least not inviting discussions about security (by choosing a different field).